src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,288 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Feature extractor class for Pop2Piano"""
+
+from typing import List, Optional, Union
+
+import librosa
+import numpy as np
+import scipy
+import torch
+from torch.nn.utils.rnn import pad_sequence
+
+from ...audio_utils import fram_wave, get_mel_filter_banks, stft
+from ...feature_extraction_sequence_utils import SequenceFeatureExtractor
+from ...feature_extraction_utils import BatchFeature
+from ...utils import OptionalDependencyNotAvailable, TensorType, is_essentia_available, logging
+
+
+try:
+    if not is_essentia_available:
+        raise OptionalDependencyNotAvailable()
+except ImportError:
+    raise ImportError("There was an error while importing essentia!")
+else:
+    import essentia
+    import essentia.standard
+
+
+logger = logging.get_logger(__name__)
+
+
+class Pop2PianoFeatureExtractor(SequenceFeatureExtractor):
+    r"""
+    Constructs a Pop2Piano feature extractor.
+
+    This feature extractor inherits from [`~feature_extraction_sequence_utils.SequenceFeatureExtractor`] which contains
+    most of the main methods. Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+    This class extracts rhythm and does preprocesses before being passed through the transformer model.
+        sampling_rate (`int`, *optional*, defaults to 22050):
+            Target Sampling rate of audio signal. It's the sampling rate that we forward to the model.
+        padding_value (`int`, *optional*, defaults to 0):
+            Padding value used to pad the audio. Should correspond to silences.
+        fft_window_size (`int`, *optional*, defaults to 4096):
+            Size of the window om which the Fourier transform is applied.
+        hop_length (`int`, *optional*, defaults to 1024):
+            Step between each window of the waveform.
+        frequency_min (`float`, *optional*, defaults to 10.0):
+            Minimum frequency.
+        nb_mel_filters (`int`, *optional*, defaults to 512):
+            Number of Mel filers to generate.
+        n_bars (`int`, *optional*, defaults to 2):
+            Determines interval between each sequence.
+    """
+    model_input_names = ["input_features"]
+
+    def __init__(
+        self,
+        sampling_rate: int = 22050,
+        padding_value: int = 0,
+        fft_window_size: int = 4096,
+        hop_length: int = 1024,
+        frequency_min: float = 10.0,
+        nb_mel_filters: int = 512,
+        n_bars: int = 2,
+        feature_size=None,
```

Comment:
Are you sure `feature_size` should be none? Typically this property is what stores the number of mel filters (i.e. we actually use `feature_size` instead of `nb_mel_filters`).